app/ante.go

```diff
@@ -76,83 +68,48 @@ func (app *mxwApp) NewAnteHandler() sdkTypes.AnteHandler {
 				return ctx, checkFeeErr
 			}
 		}
-		signerAddrs := stdTx.GetSigners()
-		if len(signerAddrs) != 1 {
-			return ctx, sdkTypes.ErrInternal(fmt.Sprintf("MXW transactions accept only one signer. it has %v senders", len(signerAddrs)))
-		}
 
-		stdSigs := stdTx.Signatures
-		if signerAcc.GetMultiSig() == nil {
-			if len(stdSigs) != 1 {
-				return ctx, sdkTypes.ErrInternal(fmt.Sprintf("MXW transactions accept only one signature. it has %v signatures", len(stdSigs)))
-			}
-		}
-
-		signerAcc, err := sdkAuth.GetSignerAcc(ctx, app.accountKeeper, signerAddrs[0])
+		// -------------------------------------------------------------------------
+		// Check signatures and updat sequence and public_keys (if it hasn't set yet)
+		// It may update account's PublicKey
+		acc, err := utils.CheckTxSig(ctx, stdTx, app.accountKeeper, app.kycKeeper)
 		if err != nil {
 			return ctx, err
 		}
 
-		// removing this condition will cause app-hash change
-		if !stdTx.Fee.Amount.IsZero() {
-			err = sdkAuth.DeductFees(app.supplyKeeper, ctx, signerAcc, stdTx.Fee.Amount)
-			if err != nil {
-				return ctx, err
-			}
-
-			signerAcc = app.accountKeeper.GetAccount(ctx, signerAcc.GetAddress())
-		}
-
-		if signerAcc.IsMultiSig() {
-			signerMultiSig := signerAcc.GetMultiSig()
-			txID, validate := signerMultiSig.ValidateMultiSigTx(stdTx)
-			if !validate {
-				return ctx, sdkTypes.ErrUnknownAddress("Invalid multisig tx.")
-			}
-
-			isMetric := signerMultiSig.IsMetric(txID)
+		// Try to delete it from pending list
+		if acc.IsMultiSig() {
+			multisig := acc.GetMultiSig()
+			txID, _ := multisig.CheckTx(stdTx)
+			isMetric := multisig.IsMetric(txID)
 			if !isMetric {
 				return ctx, sdkTypes.ErrUnknownRequest("Multisig Transaction is not valid.")
 			}
-
-			for _, v := range stdSigs {
-				accAddress, err := sdkTypes.AccAddressFromHex(v.PubKey.Address().String())
-				if err != nil {
-					return ctx, err
-				}
-				if !signerMultiSig.IsSigner(accAddress) {
-					return ctx, sdkTypes.ErrUnauthorized("Invalid multisig account signer.")
-				}
-
-				// Signer is one of the signer in groupAccount
-				signer := app.accountKeeper.GetAccount(ctx, accAddress)
-
-				// signerAcc is groupAccount
-				signBytes := types.GetSignBytes(ctx, stdTx, signerAcc)
-				_, err = processSig(ctx, signer, v, signBytes, simulate, params)
-				if err != nil {
-					return ctx, err
-				}
-			}
 			// after validating everything, delete the pendingTx
-			isDeleted := signerMultiSig.RemoveTx(txID)
+			isDeleted := multisig.RemoveTx(txID)
 			if !isDeleted {
 				return ctx, sdkTypes.ErrUnknownRequest("Delete failed.")
 			}
-			signerAcc.SetMultiSig(signerMultiSig)
-			app.accountKeeper.SetAccount(ctx, signerAcc)
-		} else {
+			acc.SetMultiSig(multisig)
+		}
 
-			signBytes := types.GetSignBytes(ctx, stdTx, signerAcc)
+		if err := acc.SetSequence(acc.GetSequence() + 1); err != nil {
```

Comment:
how do u make sure you increased the correct account sequence? For multisig account the sequence of the group address will be increased.

Reply:
If the account is signer, the sequence will increase. We get the signer information from message interface.